File: contain_duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
const containsDuplicate = nums=>{
   const hash = {};

   for(let i = 0 ; i<nums.length; i++){
        const val = nums[i]
        if (hash[val] != null || hash[val] !=  undefined) 
            return true
        hash[val] = val
   }
   return false
    
};

File: contain_duplicate.rs
use::std::collections::HashMap;
impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        let mut hash = HashMap::new();
        for num in nums.iter(){
            match hash.get(num){
                Some(x)=>return true,
                None => hash.insert(num,num)
            };
        }
        return false;
    }
}
